provisioner/clusterpy.go

```diff
@@ -212,10 +237,14 @@ func (p *clusterpyProvisioner) Provision(cluster *api.Cluster, channelConfig *ch
 
 			sort.Sort(api.NodePools(nodePools))
 			for _, nodePool := range nodePools {
-				err := updater.Update(context.Background(), nodePool)
+				err := updater.Update(ctx, nodePool)
 				if err != nil {
 					return err
 				}
+
+				if err = ctx.Err(); err != nil {
```

Comment:
Isn't this superflous? When the context was cancelled wouldn't `updater.Update(ctx, nodePool)` right above already return a non-nil `err`?

Reply:
True, but the performance of the check is rather negligible and it's easier to just keep them between all major steps on every level.